// @seven.io/api/dist/constants/byEndpoint/lookup/index.js
"use strict";
exports.__esModule = true;
exports.STRING_RESPONSE_CODES = exports.ROAMING_STATUS_CODES = exports.HLR_VALID_NUMBER_CODES = exports.HLR_STATUS_MESSAGE_CODES = exports.HLR_ROAMING_CODES = exports.HLR_REACHABLE_CODES = exports.HLR_PORTED_CODES = exports.HLR_LOOKUP_OUTCOME_CODES = exports.CNAM_API_CODES = exports.API_RESPONSE_CODES = void 0;
var ErrorCode_1 = require("../../ErrorCode");
exports.API_RESPONSE_CODES = [
    100,
    ErrorCode_1.ErrorCode.InvalidRecipientNumber,
    ErrorCode_1.ErrorCode.InsufficientCredits,
    ErrorCode_1.ErrorCode.CarrierFailed,
];
exports.CNAM_API_CODES = [
    ErrorCode_1.ErrorCode.InvalidRecipientNumber,
    ErrorCode_1.ErrorCode.CarrierFailed
];
exports.HLR_LOOKUP_OUTCOME_CODES = [1, 2, 0];
exports.HLR_PORTED_CODES = [
    'unknown',
    'ported',
    'not_ported',
    'assumed_not_ported',
    'assumed_ported'
];
exports.HLR_REACHABLE_CODES = [
    'unknown',
    'reachable',
    'undeliverable',
    'absent',
    'bad_number',
    'blacklisted'
];
exports.HLR_ROAMING_CODES = ['not_roaming',];
exports.HLR_STATUS_MESSAGE_CODES = ['error', 'success',];
exports.HLR_VALID_NUMBER_CODES = ['unknown', 'valid', 'not_valid'];
exports.ROAMING_STATUS_CODES = ['not_roaming', 'roaming', 'unknown'];
exports.STRING_RESPONSE_CODES = [
    '100',
    ErrorCode_1.ErrorCode.InvalidRecipientNumber.toString(),
    ErrorCode_1.ErrorCode.InsufficientCredits.toString(),
    ErrorCode_1.ErrorCode.CarrierFailed.toString(),
];
