// @seven.io/api/dist/Sms77Client.umd.js
(function(f){if(typeof exports==="object"&&typeof module!=="undefined"){module.exports=f()}else if(typeof define==="function"&&define.amd){define([],f)}else{var g;if(typeof window!=="undefined"){g=window}else if(typeof global!=="undefined"){g=global}else if(typeof self!=="undefined"){g=self}else{g=this}g.Sms77Client = f()}})(function(){var define,module,exports;return (function(){function r(e,n,t){function o(i,f){if(!n[i]){if(!e[i]){var c="function"==typeof require&&require;if(!f&&c)return c(i,!0);if(u)return u(i,!0);var a=new Error("Cannot find module '"+i+"'");throw a.code="MODULE_NOT_FOUND",a}var p=n[i]={exports:{}};e[i][0].call(p.exports,function(r){var n=e[i][1][r];return o(n||r)},p,p.exports,r,e,n,t)}return n[i].exports}for(var u="function"==typeof require&&require,i=0;i<t.length;i++)o(t[i]);return o}return r})()({1:[function(require,module,exports){
module.exports = require('../src/Sms77Client').default;
},{"../src/Sms77Client":3}],2:[function(require,module,exports){
"use strict";
var __assign = (this && this.__assign) || function () {
    __assign = Object.assign || function(t) {
        for (var s, i = 1, n = arguments.length; i < n; i++) {
            s = arguments[i];
            for (var p in s) if (Object.prototype.hasOwnProperty.call(s, p))
                t[p] = s[p];
        }
        return t;
    };
    return __assign.apply(this, arguments);
};
var __awaiter = (this && this.__awaiter) || function (thisArg, _arguments, P, generator) {
    function adopt(value) { return value instanceof P ? value : new P(function (resolve) { resolve(value); }); }
    return new (P || (P = Promise))(function (resolve, reject) {
        function fulfilled(value) { try { step(generator.next(value)); } catch (e) { reject(e); } }
        function rejected(value) { try { step(generator["throw"](value)); } catch (e) { reject(e); } }
        function step(result) { result.done ? resolve(result.value) : adopt(result.value).then(fulfilled, rejected); }
        step((generator = generator.apply(thisArg, _arguments || [])).next());
    });
};
var __generator = (this && this.__generator) || function (thisArg, body) {
    var _ = { label: 0, sent: function() { if (t[0] & 1) throw t[1]; return t[1]; }, trys: [], ops: [] }, f, y, t, g;
    return g = { next: verb(0), "throw": verb(1), "return": verb(2) }, typeof Symbol === "function" && (g[Symbol.iterator] = function() { return this; }), g;
    function verb(n) { return function (v) { return step([n, v]); }; }
    function step(op) {
        if (f) throw new TypeError("Generator is already executing.");
        while (g && (g = 0, op[0] && (_ = 0)), _) try {
            if (f = 1, y && (t = op[0] & 2 ? y["return"] : op[0] ? y["throw"] || ((t = y["return"]) && t.call(y), 0) : y.next) && !(t = t.call(y, op[1])).done) return t;
            if (y = 0, t) op = [op[0] & 2, t.value];
            switch (op[0]) {
                case 0: case 1: t = op; break;
                case 4: _.label++; return { value: op[1], done: false };
                case 5: _.label++; y = op[1]; op = [0]; continue;
                case 7: op = _.ops.pop(); _.trys.pop(); continue;
                default:
                    if (!(t = _.trys, t = t.length > 0 && t[t.length - 1]) && (op[0] === 6 || op[0] === 2)) { _ = 0; continue; }
                    if (op[0] === 3 && (!t || (op[1] > t[0] && op[1] < t[3]))) { _.label = op[1]; break; }
                    if (op[0] === 6 && _.label < t[1]) { _.label = t[1]; t = op; break; }
                    if (t && _.label < t[2]) { _.label = t[2]; _.ops.push(op); break; }
                    if (t[2]) _.ops.pop();
                    _.trys.pop(); continue;
            }
            op = body.call(thisArg, _);
        } catch (e) { op = [6, e]; y = 0; } finally { f = t = 0; }
        if (op[0] & 5) throw op[1]; return { value: op[0] ? op[1] : void 0, done: true };
    }
};
var __read = (this && this.__read) || function (o, n) {
    var m = typeof Symbol === "function" && o[Symbol.iterator];
    if (!m) return o;
    var i = m.call(o), r, ar = [], e;
    try {
        while ((n === void 0 || n-- > 0) && !(r = i.next()).done) ar.push(r.value);
    }
    catch (error) { e = { error: error }; }
    finally {
        try {
            if (r && !r.done && (m = i["return"])) m.call(i);
        }
        finally { if (e) throw e.error; }
    }
    return ar;
};
var __importDefault = (this && this.__importDefault) || function (mod) {
    return (mod && mod.__esModule) ? mod : { "default": mod };
};
exports.__esModule = true;
exports.BaseClient = void 0;
var Util_1 = __importDefault(require("./lib/Util"));
var ErrorCode_1 = require("./constants/ErrorCode");
var METHODS = [
    'delete',
    'get',
    'post'
];
var BaseClient = /** @class */ (function () {
    function BaseClient(apiKey, sentWith, debug) {
        if (sentWith === void 0) { sentWith = 'js'; }
        if (debug === void 0) { debug = false; }
        var _this = this;
        this.apiKey = apiKey;
        this.sentWith = sentWith;
        this.debug = debug;
        this["delete"] = function (endpoint, payload, json) {
            if (json === void 0) { json = false; }
            return __awaiter(_this, void 0, void 0, function () { return __generator(this, function (_a) {
                switch (_a.label) {
                    case 0: return [4 /*yield*/, this.request('delete', endpoint, payload, json)];
                    case 1: return [2 /*return*/, _a.sent()];
                }
            }); });
        };
        this.get = function (endpoint, payload) { return __awaiter(_this, void 0, void 0, function () { return __generator(this, function (_a) {
            switch (_a.label) {
                case 0: return [4 /*yield*/, this.request('get', endpoint, payload)];
                case 1: return [2 /*return*/, _a.sent()];
            }
        }); }); };
        this.post = function (endpoint, payload) { return __awaiter(_this, void 0, void 0, function () { return __generator(this, function (_a) {
            switch (_a.label) {
                case 0: return [4 /*yield*/, this.request('post', endpoint, payload)];
                case 1: return [2 /*return*/, _a.sent()];
            }
        }); }); };
        this.normalizePayload = function (o) { return Object.fromEntries(Object.entries(o).map(function (_a) {
            var _b = __read(_a, 2), k = _b[0], v = _b[1];
            return [
                k,
                Util_1["default"].toNumberedBool(v)
            ];
        })); };
        this.assertIs(this);
    }
    BaseClient.prototype.assertIs = function (value) {
    };
    BaseClient.prototype.request = function (method, endpoint, o, json) {
        if (o === void 0) { o = {}; }
        if (json === void 0) { json = false; }
        return __awaiter(this, void 0, void 0, function () {
            var url, headers, opts, params_1, res, body, apiCode, type, parsed;
            return __generator(this, function (_a) {
                switch (_a.label) {
                    case 0:
                        url = "".concat(BaseClient.BASE_URL, "/").concat(endpoint);
                        headers = {
                            sentWith: this.sentWith
                        };
                        if (this.apiKey.startsWith('Bearer '))
                            headers.Authorization = this.apiKey;
                        else
                            headers['X-Api-Key'] = this.apiKey;
                        opts = {
                            headers: headers,
                            method: method
                        };
                        if (o && Object.keys(o).length) {
                            o = this.normalizePayload(o);
                            if (json) {
                                opts.body = JSON.stringify(o);
                                headers['Content-Type'] = 'application/json';
                            }
                            else {
                                params_1 = new URLSearchParams;
                                Object.entries(o).forEach((function (_a) {
                                    var _b = __read(_a, 2), k = _b[0], v = _b[1];
                                    if (Array.isArray(v)) {
                                        v.forEach(function (i, n) {
                                            if (typeof i === 'object' && 'contents' in i && 'name' in i) { // is likely SmsFile
                                                var prepend = "files[".concat(n, "]");
                                                params_1.set("".concat(prepend, "[contents]"), i.contents);
                                                params_1.set("".concat(prepend, "[name]"), i.name);
                                                if ('password' in i)
                                                    params_1.set("".concat(prepend, "[password]"), i.password);
                                                if ('validity' in i)
                                                    params_1.set("".concat(prepend, "[validity]"), i.validity);
                                            }
                                            else
                                                params_1.append(k, i);
                                        });
                                    }
                                    else
                                        params_1.set(k, v);
                                }));
                                switch (method) {
                                    case 'get':
                                        url += "?".concat(params_1.toString());
                                        break;
                                    default:
                                        opts.body = params_1;
                                }
                            }
                        }
                        return [4 /*yield*/, fetch(url, opts)];
                    case 1:
                        res = _a.sent();
                        return [4 /*yield*/, res.text()];
                    case 2:
                        body = _a.sent();
                        apiCode = null;
                        if (typeof body === 'string') {
                            try {
                                body = JSON.parse(body);
                            }
                            catch (_) {
                            }
                        }
                        type = typeof body;
                        if (type === 'string' || type === 'number') {
                            parsed = Number.parseFloat("".concat(body));
                            if (Number.isInteger(parsed))
                                apiCode = parsed;
                        }
                        if (this.debug)
                            console.debug({
                                request: __assign(__assign({}, opts), { url: url, body: opts.body instanceof URLSearchParams
                                        ? Object.fromEntries(opts.body) : opts.body }),
                                response: {
                                    apiCode: apiCode,
                                    body: body,
                                    headers: Object.fromEntries(res.headers),
                                    status: res.status
                                }
                            });
                        if (apiCode && apiCode in ErrorCode_1.ErrorCode)
                            throw new Error("".concat(apiCode, ": ").concat(ErrorCode_1.ErrorCode[apiCode]));
                        return [2 /*return*/, body];
                }
            });
        });
    };
    BaseClient.BASE_URL = 'https://gateway.sms77.io/api';
    return BaseClient;
}());
exports.BaseClient = BaseClient;

},{"./constants/ErrorCode":5,"./lib/Util":9}],3:[function(require,module,exports){
"use strict";
var __extends = (this && this.__extends) || (function () {
    var extendStatics = function (d, b) {
        extendStatics = Object.setPrototypeOf ||
            ({ __proto__: [] } instanceof Array && function (d, b) { d.__proto__ = b; }) ||
            function (d, b) { for (var p in b) if (Object.prototype.hasOwnProperty.call(b, p)) d[p] = b[p]; };
        return extendStatics(d, b);
    };
    return function (d, b) {
        if (typeof b !== "function" && b !== null)
            throw new TypeError("Class extends value " + String(b) + " is not a constructor or null");
        extendStatics(d, b);
        function __() { this.constructor = d; }
        d.prototype = b === null ? Object.create(b) : (__.prototype = b.prototype, new __());
    };
})();
var __assign = (this && this.__assign) || function () {
    __assign = Object.assign || function(t) {
        for (var s, i = 1, n = arguments.length; i < n; i++) {
            s = arguments[i];
            for (var p in s) if (Object.prototype.hasOwnProperty.call(s, p))
                t[p] = s[p];
        }
        return t;
    };
    return __assign.apply(this, arguments);
};
var __createBinding = (this && this.__createBinding) || (Object.create ? (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    var desc = Object.getOwnPropertyDescriptor(m, k);
    if (!desc || ("get" in desc ? !m.__esModule : desc.writable || desc.configurable)) {
      desc = { enumerable: true, get: function() { return m[k]; } };
    }
    Object.defineProperty(o, k2, desc);
}) : (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    o[k2] = m[k];
}));
var __exportStar = (this && this.__exportStar) || function(m, exports) {
    for (var p in m) if (p !== "default" && !Object.prototype.hasOwnProperty.call(exports, p)) __createBinding(exports, m, p);
};
var __awaiter = (this && this.__awaiter) || function (thisArg, _arguments, P, generator) {
    function adopt(value) { return value instanceof P ? value : new P(function (resolve) { resolve(value); }); }
    return new (P || (P = Promise))(function (resolve, reject) {
        function fulfilled(value) { try { step(generator.next(value)); } catch (e) { reject(e); } }
        function rejected(value) { try { step(generator["throw"](value)); } catch (e) { reject(e); } }
        function step(result) { result.done ? resolve(result.value) : adopt(result.value).then(fulfilled, rejected); }
        step((generator = generator.apply(thisArg, _arguments || [])).next());
    });
};
var __generator = (this && this.__generator) || function (thisArg, body) {
    var _ = { label: 0, sent: function() { if (t[0] & 1) throw t[1]; return t[1]; }, trys: [], ops: [] }, f, y, t, g;
    return g = { next: verb(0), "throw": verb(1), "return": verb(2) }, typeof Symbol === "function" && (g[Symbol.iterator] = function() { return this; }), g;
    function verb(n) { return function (v) { return step([n, v]); }; }
    function step(op) {
        if (f) throw new TypeError("Generator is already executing.");
        while (g && (g = 0, op[0] && (_ = 0)), _) try {
            if (f = 1, y && (t = op[0] & 2 ? y["return"] : op[0] ? y["throw"] || ((t = y["return"]) && t.call(y), 0) : y.next) && !(t = t.call(y, op[1])).done) return t;
            if (y = 0, t) op = [op[0] & 2, t.value];
            switch (op[0]) {
                case 0: case 1: t = op; break;
                case 4: _.label++; return { value: op[1], done: false };
                case 5: _.label++; y = op[1]; op = [0]; continue;
                case 7: op = _.ops.pop(); _.trys.pop(); continue;
                default:
                    if (!(t = _.trys, t = t.length > 0 && t[t.length - 1]) && (op[0] === 6 || op[0] === 2)) { _ = 0; continue; }
                    if (op[0] === 3 && (!t || (op[1] > t[0] && op[1] < t[3]))) { _.label = op[1]; break; }
                    if (op[0] === 6 && _.label < t[1]) { _.label = t[1]; t = op; break; }
                    if (t && _.label < t[2]) { _.label = t[2]; _.ops.push(op); break; }
                    if (t[2]) _.ops.pop();
                    _.trys.pop(); continue;
            }
            op = body.call(thisArg, _);
        } catch (e) { op = [6, e]; y = 0; } finally { f = t = 0; }
        if (op[0] & 5) throw op[1]; return { value: op[0] ? op[1] : void 0, done: true };
    }
};
var __importDefault = (this && this.__importDefault) || function (mod) {
    return (mod && mod.__esModule) ? mod : { "default": mod };
};
exports.__esModule = true;
exports.Sms77Client = void 0;
var BaseClient_1 = require("./BaseClient");
var Endpoint_1 = require("./constants/Endpoint");
var TextTransformer_1 = __importDefault(require("./lib/TextTransformer"));
var ContactsAction_1 = require("./constants/byEndpoint/contacts/ContactsAction");
var ContactsResponseCode_1 = require("./constants/byEndpoint/contacts/ContactsResponseCode");
__exportStar(require("./types"), exports);
var Sms77Client = /** @class */ (function (_super) {
    __extends(Sms77Client, _super);
    function Sms77Client() {
        var _this = _super !== null && _super.apply(this, arguments) || this;
        _this.analytics = function (p) { return __awaiter(_this, void 0, void 0, function () { return __generator(this, function (_a) {
            switch (_a.label) {
                case 0: return [4 /*yield*/, this.get(Endpoint_1.Endpoint.Analytics, p)];
                case 1: return [2 /*return*/, _a.sent()];
            }
        }); }); };
        _this.balance = function () { return __awaiter(_this, void 0, void 0, function () { var _a, _b; return __generator(this, function (_c) {
            switch (_c.label) {
                case 0:
                    _b = (_a = Number).parseFloat;
                    return [4 /*yield*/, this.get(Endpoint_1.Endpoint.Balance, {})];
                case 1: return [2 /*return*/, _b.apply(_a, [_c.sent()])];
            }
        }); }); };
        _this.contacts = function (p) { return __awaiter(_this, void 0, void 0, function () {
            var method, res;
            return __generator(this, function (_a) {
                switch (_a.label) {
                    case 0:
                        method = ContactsAction_1.ContactsAction.Delete === p.action ? this.post : this.get;
                        return [4 /*yield*/, method(Endpoint_1.Endpoint.Contacts, p)];
                    case 1:
                        res = _a.sent();
                        if (Array.isArray(res))
                            return [2 /*return*/, res];
                        if (typeof res === 'object' &&
                            Object.values(ContactsResponseCode_1.ContactsResponseCode).includes(Number.parseInt(res["return"]))) {
                            if (res.id)
                                return [2 /*return*/, Number.parseInt(res.id)];
                            if (undefined !== p.id)
                                return [2 /*return*/, p.id];
                        }
                        return [2 /*return*/, res];
                }
            });
        }); };
        _this.deleteSMS = function (p) { return __awaiter(_this, void 0, void 0, function () { return __generator(this, function (_a) {
            switch (_a.label) {
                case 0: return [4 /*yield*/, this["delete"](Endpoint_1.Endpoint.Sms, p, true)];
                case 1: return [2 /*return*/, _a.sent()];
            }
        }); }); };
        _this.hooks = function (p) { return __awaiter(_this, void 0, void 0, function () { return __generator(this, function (_a) {
            switch (_a.label) {
                case 0: return [4 /*yield*/, this.post(Endpoint_1.Endpoint.Hooks, p)];
                case 1: return [2 /*return*/, _a.sent()];
            }
        }); }); };
        _this.journal = function (p) { return __awaiter(_this, void 0, void 0, function () { return __generator(this, function (_a) {
            switch (_a.label) {
                case 0: return [4 /*yield*/, this.get(Endpoint_1.Endpoint.Journal, p)];
                case 1: return [2 /*return*/, _a.sent()];
            }
        }); }); };
        _this.lookup = function (p) { return __awaiter(_this, void 0, void 0, function () { return __generator(this, function (_a) {
            switch (_a.label) {
                case 0: return [4 /*yield*/, this.post(Endpoint_1.Endpoint.Lookup, p)];
                case 1: return [2 /*return*/, _a.sent()];
            }
        }); }); };
        _this.pricing = function (p) { return __awaiter(_this, void 0, void 0, function () { return __generator(this, function (_a) {
            switch (_a.label) {
                case 0: return [4 /*yield*/, this.post(Endpoint_1.Endpoint.Pricing, p)];
                case 1: return [2 /*return*/, _a.sent()];
            }
        }); }); };
        _this.sms = function (p) { return __awaiter(_this, void 0, void 0, function () { return __generator(this, function (_a) {
            switch (_a.label) {
                case 0: return [4 /*yield*/, this.post(Endpoint_1.Endpoint.Sms, p)];
                case 1: return [2 /*return*/, _a.sent()];
            }
        }); }); };
        _this.status = function (p) { return __awaiter(_this, void 0, void 0, function () {
            var res;
            return __generator(this, function (_a) {
                switch (_a.label) {
                    case 0: return [4 /*yield*/, this.get(Endpoint_1.Endpoint.Status, p)];
                    case 1:
                        res = _a.sent();
                        return [2 /*return*/, p._json ? TextTransformer_1["default"].status(res) : res];
                }
            });
        }); };
        _this.subaccounts = {
            _dynamic: function (p) { return __awaiter(_this, void 0, void 0, function () {
                var _a;
                return __generator(this, function (_b) {
                    switch (_b.label) {
                        case 0:
                            _a = p.action;
                            switch (_a) {
                                case 'read': return [3 /*break*/, 1];
                                case 'create': return [3 /*break*/, 3];
                                case 'delete': return [3 /*break*/, 5];
                                case 'update': return [3 /*break*/, 7];
                                case 'transfer_credits': return [3 /*break*/, 9];
                            }
                            return [3 /*break*/, 11];
                        case 1: return [4 /*yield*/, this.subaccounts.read()];
                        case 2: return [2 /*return*/, _b.sent()];
                        case 3: return [4 /*yield*/, this.subaccounts
                                .create(p)];
                        case 4: return [2 /*return*/, _b.sent()];
                        case 5: return [4 /*yield*/, this.subaccounts["delete"](p)];
                        case 6: return [2 /*return*/, _b.sent()];
                        case 7: return [4 /*yield*/, this.subaccounts
                                .autoCharge(p)];
                        case 8: return [2 /*return*/, _b.sent()];
                        case 9: return [4 /*yield*/, this.subaccounts
                                .transferCredits(p)];
                        case 10: return [2 /*return*/, _b.sent()];
                        case 11: throw new Error("Invalid action ".concat(p.action));
                    }
                });
            }); },
            read: function () { return __awaiter(_this, void 0, void 0, function () {
                var params;
                return __generator(this, function (_a) {
                    switch (_a.label) {
                        case 0:
                            params = {
                                action: 'read'
                            };
                            return [4 /*yield*/, this.get(Endpoint_1.Endpoint.Subaccounts, params)];
                        case 1: return [2 /*return*/, _a.sent()];
                    }
                });
            }); },
            create: function (p) { return __awaiter(_this, void 0, void 0, function () {
                var params;
                return __generator(this, function (_a) {
                    switch (_a.label) {
                        case 0:
                            params = __assign({ action: 'create' }, p);
                            return [4 /*yield*/, this.post(Endpoint_1.Endpoint.Subaccounts, params)];
                        case 1: return [2 /*return*/, _a.sent()];
                    }
                });
            }); },
            "delete": function (p) { return __awaiter(_this, void 0, void 0, function () {
                var params;
                return __generator(this, function (_a) {
                    switch (_a.label) {
                        case 0:
                            params = __assign({ action: 'delete' }, p);
                            return [4 /*yield*/, this.post(Endpoint_1.Endpoint.Subaccounts, params)];
                        case 1: return [2 /*return*/, _a.sent()];
                    }
                });
            }); },
            transferCredits: function (p) { return __awaiter(_this, void 0, void 0, function () {
                var params;
                return __generator(this, function (_a) {
                    switch (_a.label) {
                        case 0:
                            params = __assign({ action: 'transfer_credits' }, p);
                            return [4 /*yield*/, this.post(Endpoint_1.Endpoint.Subaccounts, params)];
                        case 1: return [2 /*return*/, _a.sent()];
                    }
                });
            }); },
            autoCharge: function (p) { return __awaiter(_this, void 0, void 0, function () {
                var params;
                return __generator(this, function (_a) {
                    switch (_a.label) {
                        case 0:
                            params = __assign({ action: 'update' }, p);
                            return [4 /*yield*/, this.post(Endpoint_1.Endpoint.Subaccounts, params)];
                        case 1: return [2 /*return*/, _a.sent()];
                    }
                });
            }); }
        };
        _this.validateForVoice = function (p) { return __awaiter(_this, void 0, void 0, function () {
            return __generator(this, function (_a) {
                switch (_a.label) {
                    case 0: return [4 /*yield*/, this.post(Endpoint_1.Endpoint.ValidateForVoice, p)];
                    case 1: return [2 /*return*/, _a.sent()];
                }
            });
        }); };
        _this.voice = function (p) { return __awaiter(_this, void 0, void 0, function () { return __generator(this, function (_a) {
            switch (_a.label) {
                case 0: return [4 /*yield*/, this.post(Endpoint_1.Endpoint.Voice, p)];
                case 1: return [2 /*return*/, _a.sent()];
            }
        }); }); };
        return _this;
    }
    return Sms77Client;
}(BaseClient_1.BaseClient));
exports.Sms77Client = Sms77Client;
exports["default"] = Sms77Client;

},{"./BaseClient":2,"./constants/Endpoint":4,"./constants/byEndpoint/contacts/ContactsAction":6,"./constants/byEndpoint/contacts/ContactsResponseCode":7,"./lib/TextTransformer":8,"./types":13}],4:[function(require,module,exports){
"use strict";
exports.__esModule = true;
exports.Endpoint = void 0;
var Endpoint;
(function (Endpoint) {
    Endpoint["Analytics"] = "analytics";
    Endpoint["Balance"] = "balance";
    Endpoint["Contacts"] = "contacts";
    Endpoint["Hooks"] = "hooks";
    Endpoint["Lookup"] = "lookup";
    Endpoint["Journal"] = "journal";
    Endpoint["Pricing"] = "pricing";
    Endpoint["Sms"] = "sms";
    Endpoint["Status"] = "status";
    Endpoint["Subaccounts"] = "subaccounts";
    Endpoint["ValidateForVoice"] = "validate_for_voice";
    Endpoint["Voice"] = "voice";
})(Endpoint = exports.Endpoint || (exports.Endpoint = {}));

},{}],5:[function(require,module,exports){
"use strict";
exports.__esModule = true;
exports.ErrorCode = void 0;
var ErrorCode;
(function (ErrorCode) {
    ErrorCode[ErrorCode["InvalidCountryCode"] = 201] = "InvalidCountryCode";
    ErrorCode[ErrorCode["InvalidRecipientNumber"] = 202] = "InvalidRecipientNumber";
    ErrorCode[ErrorCode["MissingAuthentication"] = 300] = "MissingAuthentication";
    ErrorCode[ErrorCode["MissingParameterTo"] = 301] = "MissingParameterTo";
    ErrorCode[ErrorCode["MissingParameterType"] = 304] = "MissingParameterType";
    ErrorCode[ErrorCode["MissingParameterText"] = 305] = "MissingParameterText";
    ErrorCode[ErrorCode["InvalidSender"] = 306] = "InvalidSender";
    ErrorCode[ErrorCode["MissingParameterUrl"] = 307] = "MissingParameterUrl";
    ErrorCode[ErrorCode["InvalidType"] = 400] = "InvalidType";
    ErrorCode[ErrorCode["ParameterLongExceedsCharLimit"] = 401] = "ParameterLongExceedsCharLimit";
    ErrorCode[ErrorCode["PreventedByReloadLock"] = 402] = "PreventedByReloadLock";
    ErrorCode[ErrorCode["InsufficientCredits"] = 500] = "InsufficientCredits";
    ErrorCode[ErrorCode["CarrierFailed"] = 600] = "CarrierFailed";
    ErrorCode[ErrorCode["UnknownError"] = 700] = "UnknownError";
    ErrorCode[ErrorCode["MissingLogoFile"] = 801] = "MissingLogoFile";
    ErrorCode[ErrorCode["NonExistingLogoFile"] = 802] = "NonExistingLogoFile";
    ErrorCode[ErrorCode["MissingRingTone"] = 803] = "MissingRingTone";
    ErrorCode[ErrorCode["InvalidApiKey"] = 900] = "InvalidApiKey";
    ErrorCode[ErrorCode["InvalidMessageId"] = 901] = "InvalidMessageId";
    ErrorCode[ErrorCode["DeactivatedApi"] = 902] = "DeactivatedApi";
    ErrorCode[ErrorCode["DisallowedIp"] = 903] = "DisallowedIp";
})(ErrorCode = exports.ErrorCode || (exports.ErrorCode = {}));

},{}],6:[function(require,module,exports){
"use strict";
exports.__esModule = true;
exports.ContactsAction = void 0;
var ContactsAction;
(function (ContactsAction) {
    ContactsAction["Delete"] = "del";
    ContactsAction["Read"] = "read";
    ContactsAction["Write"] = "write";
})(ContactsAction = exports.ContactsAction || (exports.ContactsAction = {}));

},{}],7:[function(require,module,exports){
"use strict";
exports.__esModule = true;
exports.ContactsResponseCode = void 0;
var ContactsResponseCode;
(function (ContactsResponseCode) {
    ContactsResponseCode[ContactsResponseCode["Error"] = 151] = "Error";
    ContactsResponseCode[ContactsResponseCode["Success"] = 152] = "Success";
})(ContactsResponseCode = exports.ContactsResponseCode || (exports.ContactsResponseCode = {}));

},{}],8:[function(require,module,exports){
"use strict";
var __read = (this && this.__read) || function (o, n) {
    var m = typeof Symbol === "function" && o[Symbol.iterator];
    if (!m) return o;
    var i = m.call(o), r, ar = [], e;
    try {
        while ((n === void 0 || n-- > 0) && !(r = i.next()).done) ar.push(r.value);
    }
    catch (error) { e = { error: error }; }
    finally {
        try {
            if (r && !r.done && (m = i["return"])) m.call(i);
        }
        finally { if (e) throw e.error; }
    }
    return ar;
};
var __values = (this && this.__values) || function(o) {
    var s = typeof Symbol === "function" && Symbol.iterator, m = s && o[s], i = 0;
    if (m) return m.call(o);
    if (o && typeof o.length === "number") return {
        next: function () {
            if (o && i >= o.length) o = void 0;
            return { value: o && o[i++], done: !o };
        }
    };
    throw new TypeError(s ? "Object is not iterable." : "Symbol.iterator is not defined.");
};
var __importDefault = (this && this.__importDefault) || function (mod) {
    return (mod && mod.__esModule) ? mod : { "default": mod };
};
exports.__esModule = true;
var Util_1 = __importDefault(require("./Util"));
var TextTransformer = /** @class */ (function () {
    function TextTransformer() {
    }
    TextTransformer.contacts = function (csv) {
        return Util_1["default"].csvToArray(csv, false)
            .map(function (_a) {
            var _b = __read(_a, 3), id = _b[0], nick = _b[1], number = _b[2];
            return ({
                id: Number.parseInt(id),
                nick: nick,
                number: number
            });
        });
    };
    TextTransformer.pricing = function (csv) {
        var e_1, _a;
        var networks = Util_1["default"].csvToArray(csv, true);
        var countries = [];
        var _loop_1 = function (network) {
            var _b = __read(network, 9), countryCode = _b[0], countryName = _b[1], countryPrefix = _b[2], mcc = _b[3], mncs = _b[4], networkName = _b[5], price = _b[6], features = _b[7], comment = _b[8];
            var country = countries.find(function (c) { return c.countryCode === countryCode; });
            var hasCountry = undefined !== country;
            var obj = country || {
                countryCode: countryCode,
                countryName: countryName,
                countryPrefix: countryPrefix,
                networks: []
            };
            obj.networks.push({
                comment: comment,
                features: features.split(',').filter(function (s) { return '' !== s; }),
                mcc: mcc,
                mncs: mncs.split(','),
                networkName: networkName.trim(),
                price: Number.parseFloat(price)
            });
            if (!hasCountry) {
                countries.push(obj);
            }
        };
        try {
            for (var networks_1 = __values(networks), networks_1_1 = networks_1.next(); !networks_1_1.done; networks_1_1 = networks_1.next()) {
                var network = networks_1_1.value;
                _loop_1(network);
            }
        }
        catch (e_1_1) { e_1 = { error: e_1_1 }; }
        finally {
            try {
                if (networks_1_1 && !networks_1_1.done && (_a = networks_1["return"])) _a.call(networks_1);
            }
            finally { if (e_1) throw e_1.error; }
        }
        return {
            countCountries: countries.length,
            countNetworks: networks.length,
            countries: countries
        };
    };
    TextTransformer.status = function (text) {
        var _a = __read(Util_1["default"].splitByLine(text), 2), report = _a[0], timestamp = _a[1];
        return {
            report: report,
            timestamp: timestamp
        };
    };
    return TextTransformer;
}());
exports["default"] = TextTransformer;

},{"./Util":9}],9:[function(require,module,exports){
"use strict";
exports.__esModule = true;
var Util = /** @class */ (function () {
    function Util() {
    }
    Util.addSignature = function (_a) {
        var position = _a.position, signature = _a.signature, text = _a.text;
        return 'append' === position
            ? "".concat(signature).concat(text) : "".concat(text).concat(signature);
    };
    Util.toNumberedBool = function (value) {
        return typeof value === 'boolean'
            ? value
                ? 1
                : 0
            : value;
    };
    Util.splitByLine = function (str) {
        return str.split(/[\r\n]+/);
    };
    Util.isNumeric = function (val) {
        return Number.isFinite(val) && !Number.isNaN(Number.parseFloat(val));
    };
    /* Thanks Michael @ https://stackoverflow.com/a/25352300/9737058 */
    Util.isAlphaNumeric = function (str) {
        for (var i = 0; i < str.length; i++) {
            var code = str.charCodeAt(i);
            if (!(code > 47 && code < 58) && // numeric (0-9)
                !(code > 64 && code < 91) && // upper alpha (A-Z)
                !(code > 96 && code < 123)) { // lower alpha (a-z)
                return false;
            }
        }
        return true;
    };
    Util.csvToArray = function (csv, skipHeader) {
        var arr = Util.splitByLine(csv.trim().replace(/"/g, ''));
        if (skipHeader) {
            arr.shift();
        }
        return arr.map(function (l) { return l.trim().split(';'); });
    };
    return Util;
}());
exports["default"] = Util;

},{}],10:[function(require,module,exports){
"use strict";
exports.__esModule = true;

},{}],11:[function(require,module,exports){
arguments[4][10][0].apply(exports,arguments)
},{"dup":10}],12:[function(require,module,exports){
arguments[4][10][0].apply(exports,arguments)
},{"dup":10}],13:[function(require,module,exports){
"use strict";
var __createBinding = (this && this.__createBinding) || (Object.create ? (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    var desc = Object.getOwnPropertyDescriptor(m, k);
    if (!desc || ("get" in desc ? !m.__esModule : desc.writable || desc.configurable)) {
      desc = { enumerable: true, get: function() { return m[k]; } };
    }
    Object.defineProperty(o, k2, desc);
}) : (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    o[k2] = m[k];
}));
var __exportStar = (this && this.__exportStar) || function(m, exports) {
    for (var p in m) if (p !== "default" && !Object.prototype.hasOwnProperty.call(exports, p)) __createBinding(exports, m, p);
};
exports.__esModule = true;
__exportStar(require("./analytics"), exports);
__exportStar(require("./contacts"), exports);
__exportStar(require("./hooks"), exports);
__exportStar(require("./journal"), exports);
__exportStar(require("./lookup"), exports);
__exportStar(require("./pricing"), exports);
__exportStar(require("./sms"), exports);
__exportStar(require("./status"), exports);
__exportStar(require("./subaccounts"), exports);
__exportStar(require("./validateForVoice"), exports);
__exportStar(require("./voice"), exports);

},{"./analytics":10,"./contacts":11,"./hooks":12,"./journal":14,"./lookup":15,"./pricing":16,"./sms":17,"./status":18,"./subaccounts":19,"./validateForVoice":20,"./voice":21}],14:[function(require,module,exports){
arguments[4][10][0].apply(exports,arguments)
},{"dup":10}],15:[function(require,module,exports){
arguments[4][10][0].apply(exports,arguments)
},{"dup":10}],16:[function(require,module,exports){
arguments[4][10][0].apply(exports,arguments)
},{"dup":10}],17:[function(require,module,exports){
arguments[4][10][0].apply(exports,arguments)
},{"dup":10}],18:[function(require,module,exports){
arguments[4][10][0].apply(exports,arguments)
},{"dup":10}],19:[function(require,module,exports){
arguments[4][10][0].apply(exports,arguments)
},{"dup":10}],20:[function(require,module,exports){
arguments[4][10][0].apply(exports,arguments)
},{"dup":10}],21:[function(require,module,exports){
arguments[4][10][0].apply(exports,arguments)
},{"dup":10}]},{},[1])(1)
});
