// sms77-client/dist/constants/ErrorCode.js
"use strict";
exports.__esModule = true;
exports.ErrorCode = void 0;
var ErrorCode;
(function (ErrorCode) {
    ErrorCode[ErrorCode["InvalidCountryCode"] = 201] = "InvalidCountryCode";
    ErrorCode[ErrorCode["InvalidRecipientNumber"] = 202] = "InvalidRecipientNumber";
    ErrorCode[ErrorCode["MissingAuthentication"] = 300] = "MissingAuthentication";
    ErrorCode[ErrorCode["MissingParameterTo"] = 301] = "MissingParameterTo";
    ErrorCode[ErrorCode["MissingParameterType"] = 304] = "MissingParameterType";
    ErrorCode[ErrorCode["MissingParameterText"] = 305] = "MissingParameterText";
    ErrorCode[ErrorCode["InvalidSender"] = 306] = "InvalidSender";
    ErrorCode[ErrorCode["MissingParameterUrl"] = 307] = "MissingParameterUrl";
    ErrorCode[ErrorCode["InvalidType"] = 400] = "InvalidType";
    ErrorCode[ErrorCode["ParameterLongExceedsCharLimit"] = 401] = "ParameterLongExceedsCharLimit";
    ErrorCode[ErrorCode["PreventedByReloadLock"] = 402] = "PreventedByReloadLock";
    ErrorCode[ErrorCode["InsufficientCredits"] = 500] = "InsufficientCredits";
    ErrorCode[ErrorCode["CarrierFailed"] = 600] = "CarrierFailed";
    ErrorCode[ErrorCode["UnknownError"] = 700] = "UnknownError";
    ErrorCode[ErrorCode["MissingLogoFile"] = 801] = "MissingLogoFile";
    ErrorCode[ErrorCode["NonExistingLogoFile"] = 802] = "NonExistingLogoFile";
    ErrorCode[ErrorCode["MissingRingTone"] = 803] = "MissingRingTone";
    ErrorCode[ErrorCode["InvalidApiKey"] = 900] = "InvalidApiKey";
    ErrorCode[ErrorCode["InvalidMessageId"] = 901] = "InvalidMessageId";
    ErrorCode[ErrorCode["DeactivatedApi"] = 902] = "DeactivatedApi";
    ErrorCode[ErrorCode["DisallowedIp"] = 903] = "DisallowedIp";
})(ErrorCode = exports.ErrorCode || (exports.ErrorCode = {}));
